hoverxref/domains.py

```diff
@@ -45,6 +45,18 @@ def _inject_hoverxref_data(self, env, refnode, typ, docname, docpath, labelid):
             'data-docpath': docpath,
             'data-section': labelid,
         }
+        url = refnode.get('refuri')
+        if url:
+            refnode._hoverxref.update({
+                # FIXME: data-url requires to use the full URL here. At this
+                # point, we need to know the domain where the project is hosted
+                'data-url': f'https://sphinx-hoverxref--146.org.readthedocs.build/en/146/{url}',
+            })
+        else:
+            logger.info(
+                'refuri not found for node. node=%s',
+                refnode.__dict__,
+            )
```

Comment:
Most of this code could probably be deleted. We don't really need to generate the URL in the backend and we can grab it from `element.href` (see https://github.com/readthedocs/sphinx-hoverxref/pull/146/files#diff-3e695bb338248f5f60a4384f4240bbfbccb6b274106166954b6643a58e655455R181)

In fact, in some opportunities, for some reason, Sphinx does not know the URL at this point (e.g. autoapi nodes) and the `data-url` ends up being empty. In those cases, the current implementation is using `element.href` and it's working. So, I think we should use that way for all the nodes and remove this Python code.

Also note that Python does not know where the documentation is hosted (domain), so it can't generate the full URL upfront. I hardcoded it for now.

@ericholscher I plan to remove all this Python code and refactor some places. Do you have any opinion here that would block this new implementation?

Reply:
I went ahead an removed everything that's not required anymore and fixed the tests in another branch just in case, this commit: 2f0d31acb

Reply:
Yea, I don't think there's any value in setting the URL twice. We should just use the URL that is in the HTML via JS.